### PR TITLE
feat: create login page

### LINKED ISSUE #43: Login page

### PR DESCRIPTION
**Pull request by banahaker**
# [ NEW FEATURE ] Login Page

create login page for lipoic web

Fixes #43 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by banahaker** [comment]
LOG: daloa @a3510377 write a perfect code

**Comment by banahaker** [comment]
i built the base, then i'll  better them, i'm going on!

File: src/components/BlockSignUpPage.vue [hunk 1]
```diff
@@ -0,0 +1 @@
+<template>sign up</template>
```
[the remark this answers is not in the document]
Comment (banahaker) [reply src/components/BlockSignUpPage.vue]: resolved !

File: src/MainApp.vue [hunk 2]
```diff
@@ -1,4 +1,18 @@
-<script setup lang="ts"></script>
+<script setup lang="ts">
+import { onMounted, onUnmounted } from 'vue';
+
+const setStyle = (key: `--${string}`, value: string | null) =>
```
Comment (SiongSng) [inline src/MainApp.vue:4]: Please add comments

File: src/components/BlockLoginPage.vue [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+const loginFormData = reactive<loginData>({
+  username: '',
+  password: '',
+  stayLoggedIn: false,
+});
+
+const oauthButtons = [
+  { title: 'Google', img: 'login-google' },
+  { title: 'FaceBook', img: 'login-facebook' },
+  { title: '臺灣雲端教育雲帳號', img: 'login-taiwanOpenId' },
```
Comment (SiongSng) [inline src/components/BlockLoginPage.vue]: Missing i18n

File: src/scss/global.scss [hunk 4]
```diff
@@ -13,9 +13,10 @@ $White: #ffffff;
 $BrightGrey: #555762;
 $LightBlack: #4f4f4f;
 $MainColor: #5efff2;
+$MainPurple: #7b6ff6;
 
 $MainColor: $MainColor;
```
Comment (SiongSng) [inline src/scss/global.scss]:
```suggestion
```

File: src/locales/zh-CN.json [hunk 5]
```diff
@@ -9,7 +9,17 @@
   "footer.copyright": "版权所有 © 2022 Lipoic. 保留所有权利。",
   "home.subtitle.1": "专门为师生们设计的多功能教育平台",
   "home.subtitle.2": "熱愛開放原始碼與開放教育精神",
+  "auth.login.welcome": "欢迎来到 Lipoic",
   "home.login.student": "学生",
   "home.login.teacher": "老师",
-  "home.login.parents": "家长"
+  "home.login.parents": "家长",
+  "auth.login.title": "登入",
+  "auth.login.usernameInput": "使用者名称",
```
Comment (SiongSng) [inline src/locales/zh-CN.json]:
```suggestion
  "auth.login.usernameInput": "用户名称",
```

**Review by SiongSng — CHANGES_REQUESTED** [review]

File: src/components/BlockLoginPage.vue (hunk 3 above)
Comment (banahaker) [reply src/components/BlockLoginPage.vue]: 1. why `Google` and `FaceBook` need i18n?
2. why `臺灣雲端教育帳號` need a English translate or is it have one?

Comment (SiongSng) [reply src/components/BlockLoginPage.vue]: Taiwan Cloud Education?

**Review by SiongSng — APPROVED** [review]
LGTM!

**Pull request merged**